[Janelle Loh] iP

janelleljt opened the pull request:
## BobTheBot
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

BobTheBob, friendly task managing bot, frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://nus-cs2103ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

janelleljt committed:
implemeted skeletal version of Bob the Bot

janelleljt committed:
added ability to store text entered by user and display back to the user when requested

janelleljt committed:
added ability to mark tasks as done and undone

janelleljt committed:
added support for tracking 3 different types of tasks

janelleljt committed:
taught bob the bot to deal with errors such as incorrect inputs by users, added DukeException

janelleljt committed:
added support for deleting tasks from the list

janelleljt committed:
wrote tests for Bob the Bot using I/O redirection technique

janelleljt committed:
changed DukeException to BobException

janelleljt committed:
made changes to excpetion handling

janelleljt committed:
write test for text UI testing

janelleljt committed:
store tasks in hard disk automatically whenever task list changes

janelleljt committed:
latest updated version of master branch

janelleljt committed:
latest updated version of master branch

janelleljt committed:
Revert "latest updated version of master branch"

This reverts commit f434204bc702f1d1f9943b2c6c032e9188aa248f.

janelleljt committed:
Merge branch 'branch-Level-7'

janelleljt committed:
resolve merge conflict

janelleljt committed:
Refactor code to extract out more closely related OOP classes

janelleljt committed:
organise classes into suitable java packages

janelleljt committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

janelleljt committed:
set up Gradle to build and run BobTheBot

janelleljt committed:
Add JUnit tests to test the behavior of the code

janelleljt committed:
Package app as JAR file

janelleljt committed:
Added JavaDocs

janelleljt committed:
Edit code to be in line with coding standard

janelleljt committed:
latest version of master branch to merge

janelleljt committed:
Resolve merge conflict cause by attempting to merge master branch with branch-A-JavaDoc

janelleljt committed:
Edit code to be in line with coding standard

janelleljt committed:
Remove unnecessary lines of code

janelleljt committed:
Alter bot to handle find command

janelleljt committed:
Resolve merge conflict cause by attempting to merge master branch with branch-Level-9

janelleljt committed:
Update code to be in line with code standards

seelengxd commented on src/main/java/bobthebot/utils/Ui.java:
```diff
@@ -0,0 +1,132 @@
+     * Method which says goodbye to the user.
+     * @param list ToDoList of the items the user still has to accomplish.
+     * */
+    public static void goodbye(ToDoList list) {
```
```suggestion
    public static void sayGoodbye(ToDoList list) {
```
Perhaps this can be slightly changed from a noun to be a verb since it is a method?

seelengxd commented on src/main/java/bobthebot/tasks/Event.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    /**
+     * Method which converts an event to a format for storage.
```
```suggestion
     * Converts an event to a format for storage.
```
Should this be a word ending with `s`? ([relevant coding standard](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=Write%20the%20first,or%20Returnning%20etc.) - in other methods as well)

seelengxd commented on src/main/java/bobthebot/bob/BobTheBot.java:
```diff
@@ -0,0 +1,55 @@
+    private Parser parser;
+
+    /**
+     * Constructor for instance of BobTheBot
```
```suggestion
     * Constructor for instance of BobTheBot.
```

Should this have punctuation? [relevant coding standard](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=Punctuation%20behind%20each%20parameter%20description)

seelengxd commented on src/main/java/bobthebot/tasks/Task.java:
```diff
@@ -0,0 +1,60 @@
+
+/**
+ * Class representing tasks.
+ * */
```
```suggestion
 *
 */
```
[relevant coding standard - subsequent * should be aligned with the first one](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=Subsequent%20*%20is%20aligned%20with%20the%20first%20one)

seelengxd commented on src/main/java/bobthebot/utils/Ui.java:
```diff
@@ -0,0 +1,132 @@
+     * Method which formats and prints the message to the user.
+     * @param s The String to be formatted.
+     * */
+    public static void formatMessage(String s) {
```
```suggestion
    public static void formatMessage(String message) {
```
Should this be changed to a more descriptive variable name? (one letter names are only for iterators)

seelengxd reviewed (commented):
I think your code mostly follows the coding standard and naming conventions with a few exceptions (mostly javadocs and one letter name issues). I liked how you separated out the messages into their own methods for abstraction.

janelleljt committed:
Edit javadocs to be in line with Java coding standard

janelleljt committed:
Add new branch to implement GUI

janelleljt committed:
Add comments

EmilyOng commented on src/main/java/bobthebot/tasks/ToDoList.java:
```diff
@@ -0,0 +1,236 @@
+                    LocalDateTime currDate = LocalDateTime.now();
+                    DateTimeFormatter format = DateTimeFormatter.ofPattern("uuuu-MM-dd kkmm");
+                    LocalDateTime deadlineDate = LocalDateTime.parse(at, format);
+                    Boolean isAfter = deadlineDate.isAfter(currDate);
```
```suggestion
                    Boolean isDeadlineAfterCurrentDate = deadlineDate.isAfter(currDate);

```

Perhaps it might be better to have a more explicit variable naming

EmilyOng commented on src/main/java/bobthebot/utils/Storage.java:
```diff
@@ -0,0 +1,126 @@
+            try {
+                file.createNewFile();
+            } catch (IOException e) {
+                System.err.println("Not sure what an IOException is but it has occurred.");
```
It would be great to put these error strings as constants at the beginning of a class to improve readability. For example:

```java
public class Storage {
    private static final ERROR_LOAD_FILE = "Not sure what an IOException is but it has occurred.";
    ...
}
```

EmilyOng commented on src/main/java/bobthebot/utils/Storage.java:
```diff
@@ -0,0 +1,126 @@
+        }
+
+        try {
+            FileWriter fw = new FileWriter(this.filePath, false);
```
```suggestion
            FileWriter fw = new FileWriter(this.filePath, isAppendMode);

```

I think it would be awesome to use an explicit variable like `isAppendMode` to avoid "magic" values.

EmilyOng commented on src/main/java/bobthebot/tasks/Event.java:
```diff
@@ -0,0 +1,63 @@
+public class Event extends Task {
+    private String at;
+
+    /**
```
I like your clear JavaDocs!

farisanadia reviewed (commented):
Overall, I found your code easy to read for the most part except that some strings could be initialized instead of repeated (e.g., for IOException). Your javadocs is very clear and concise. Nice job!

janelleljt committed:
Add GUI to BobTheBot

janelleljt committed:
Merge branch 'branch-Level-10'

janelleljt committed:
Edit code to be in line with coding standard

janelleljt committed:
Add assertions

Assertions are needed to define assumptions about the program state,
assetions failures help to indicate if there is a bug in the code.

Let's add assertions to Storage.java to check that tasks are not null,
and at BobTheBot.java to check that bot will only enter exit command
when "bye" is entered by the user.

janelleljt committed:
Refactor Parser class to have shorter methods

janelleljt committed:
Add parser method and command class for each user input

Improve code quality in 3 ways:
1. Refactor code to have different methods for parsing
   different user input.
2. Add commands to handle the result of the use input on
   the ToDoList.

Having different methods to handle the parsing and output of
user input helps to reduce method length and leads to cleaner
code quality.

We should have a parser method to handle each input, and a
command class which handles how the user input interacts with
the ToDoList class.

janelleljt committed:
Merge pull request #2 from janelleljt/branch-A-Assertions

Add assertions

janelleljt committed:
Merge branch 'master' into branch-A-CodeQuality

janelleljt committed:
Merge pull request #3 from janelleljt/branch-A-CodeQuality

Improve code quality

janelleljt committed:
Add reminder feature

janelleljt committed:
Merge pull request #4 from janelleljt/branch-B-Reminders

Add reminder feature

janelleljt committed:
Add JavaDocs for all methods, Utilise SLAP Principle

janelleljt committed:
Add tests for Parser functions

janelleljt committed:
Add user guide to README.md

janelleljt committed:
Update user guide in README.md

janelleljt committed:
Update code to be in line with checkstyle requirements

janelleljt committed:
Update variable and method names to improve code quality

janelleljt committed:
Update GUI code to improve code quiality

janelleljt committed:
Made changed to GUI

janelleljt committed:
Make successful jar file

janelleljt committed:
Add yassified personality to bot

janelleljt committed:
Fix bestie.jar to be able to run on mac and linux
